projects/components/src/table/cells/data-renderers/table-data-cell-renderer.component.ts

```diff
@@ -85,32 +84,37 @@ export class TableDataCellRendererComponent implements OnInit {
 
     // Dynamic Component Setup
     this.cellRenderer.createComponent(
-      this.componentFactoryResolver.resolveComponentFactory(this.columnConfig.renderer!),
+      this.componentFactoryResolver.resolveComponentFactory(this.columnConfig.renderer),
       0,
-      this.tableCellRendererLookupService.createInjector(
-        this.columnConfig,
-        this.index,
-        this.cellData,
-        this.rowData,
-        this.injector
-      )
+      createInjector(this.columnConfig, this.index, this.cellData, this.rowData, this.injector)
     );
 
     // Allow columnConfig to override default alignment for cell renderer
-    this.alignment = this.columnConfig.alignment ?? this.columnConfig.renderer!.alignment;
+    this.alignment = this.columnConfig.alignment ?? this.columnConfig.renderer.alignment;
     this.leftAlignFilterButton = this.alignment === TableCellAlignmentType.Right;
   }
 
   public onClick(event: MouseEvent): void {
-    const hasClickHandler = this.columnConfig && this.columnConfig.onClick;
+    if (this.columnConfig === undefined || this.rowData === undefined) {
+      throw Error(`Undefined columnConfig or rowData`);
+    }
 
-    if (hasClickHandler) {
-      this.columnConfig!.onClick!(this.rowData!, this.columnConfig!);
+    if (this.isClickHandlerDefined(this.columnConfig.onClick)) {
+      this.columnConfig.onClick(this.rowData, this.columnConfig);
```

Comment:
Can we just use the safe access rather than the runtime type check? Is there a scenario where this could be anything besides undefined or a function? `this.columnConfig?.onClick(this.rowData, this.columnConfig)`

Reply:
I simplified it so we don't need the ts ignore comment.